rskj-core/src/test/java/co/rsk/peg/federation/FederationSupportImplTest.java

```diff
@@ -2194,6 +2195,186 @@ void save_callsStorageProviderSave() {
         verify(storageProvider).save(federationMainnetConstants.getBtcParams(), activations);
     }
 
+    @Nested
+    @Tag("commit federation")
+    @TestInstance(TestInstance.Lifecycle.PER_CLASS)
+    class CommitFederationTests {
+        private ActivationConfig.ForBlock activations;
+
+        private FederationStorageProvider federationStorageProvider;
+        private FederationSupportImpl federationSupport;
+        private PendingFederation pendingFederation;
+        private BridgeEventLogger bridgeEventLogger;
+        
+        private Block rskExecutionBlock;
+        long rskExecutionBlockNumber = 1000L;
+        long rskExecutionBlockTimestamp = 10L;
+
+        @BeforeEach
+        void setUp() {
+            StorageAccessor bridgeStorageAccessor = new InMemoryStorage();
+            federationStorageProvider = new FederationStorageProviderImpl(bridgeStorageAccessor);
+            
+            pendingFederation = new PendingFederation(FederationTestUtils.getFederationMembers(5));
+            federationStorageProvider.setPendingFederation(pendingFederation);
+
+            bridgeEventLogger = mock(BridgeEventLogger.class);
+
+            rskExecutionBlock = mock(Block.class);
+            when(rskExecutionBlock.getNumber()).thenReturn(rskExecutionBlockNumber);
+            when(rskExecutionBlock.getTimestamp()).thenReturn(rskExecutionBlockTimestamp);
+        }
+
+        @Test
+        void commitFederationAccordingToActivations_preRSKIP186_shouldCallLegacyCommitFederationAndNotPreserveFederationChangeInfo() {
+            // arrange
+            activations = ActivationConfigsForTest.papyrus200().forBlock(0L);
+            federationSupport = new FederationSupportImpl(federationMainnetConstants, federationStorageProvider, rskExecutionBlock, activations);
+            List<UTXO> utxosToMove = new ArrayList<>(federationStorageProvider.getNewFederationBtcUTXOs(federationMainnetConstants.getBtcParams(), activations));
+
+            // act
+            FederationChangeResponseCode commitFederationCall = federationSupport.commitFederationAccordingToActivations(false, pendingFederation.getHash(), bridgeEventLogger);
+
+
+            // assertions
+
+            assertUTXOsWereMoved(utxosToMove);
+
+            assertNewAndOldFederationsWereSet();
+
+            assertPendingFederationVotingWasCleaned();
+
+            assertFederationChangeInfoWasNotSet();
+
+            Federation oldFederation = federationStorageProvider.getOldFederation(federationMainnetConstants, activations);
+            Federation newFederation = federationStorageProvider.getNewFederation(federationMainnetConstants, activations);
+            assertLogCommitFederation(oldFederation, newFederation);
+
+            assertCallWasSuccessful(commitFederationCall);
```

Comment:
I like this! Using methods that abstract each kind of assertion for the test. Well done! The test looks very clear.

Reply:
thanks! it was the idea <3